static/skywire-manager-src/src/app/components/vpn/vpn-helpers.ts

```diff
@@ -0,0 +1,515 @@
+  private static currentPk = '';
+  /**
+   * Sets the Pk of the local Skywire visor. Must be called for the vpnTabsData property to
+   * work correctly.
+   */
+  static changeCurrentPk(pk: string): void {
+    this.currentPk = pk;
+  }
+
+  /**
+   * Allows to set the default tab that should be openned when entering to the server list.
+   * It is saved while the tab is openned.
+   */
+  static setDefaultTabForServerList(tab: Lists) {
+    sessionStorage.setItem(VpnHelpers.serverListTabStorageKey, tab);
+  }
+
+  /**
+   * Data for configuring the tab-bar shown in the header of the vpn client pages.
+   */
+  static get vpnTabsData(): TabButtonData[] {
+    const lastServerListTab = sessionStorage.getItem(VpnHelpers.serverListTabStorageKey);
+
+    return [
+      {
+        icon: 'power_settings_new',
+        label: 'vpn.start',
+        linkParts: ['/vpn', this.currentPk, 'status'],
+      },
+      {
+        icon: 'list',
+        label: 'vpn.servers',
+        linkParts: lastServerListTab ? ['/vpn', this.currentPk, 'servers', lastServerListTab, '1'] : ['/vpn', this.currentPk, 'servers'],
+      },
+      {
+        icon: 'settings',
+        label: 'vpn.settings',
+        linkParts: ['/vpn', this.currentPk, 'settings'],
+      },
+    ];
+  }
+
+  /**
+   * Gets the name of the translatable var that must be used for showing a latency value. This
+   * allows to add the correct measure suffix.
+   */
+  static getLatencyValueString(latency: number): string {
+    if (latency < 1000) {
+      return 'time-in-ms';
+    }
+
+    return 'time-in-segs';
+  }
+
+  /**
+   * Gets the string value to show in the UI a latency value, with an adecuate number of decimals.
+   * This function converts the value from ms to segs, if appropriate, so the value must be shown
+   * using the var returned by getLatencyValueString.
+   */
+  static getPrintableLatency(latency: number): string {
+    if (latency < 1000) {
+      return latency + '';
+    }
+
+    return (latency / 1000).toFixed(1);
+  }
+
+  /**
+   * Changes the server and connects to it. One, and only one, of the newServer params must
+   * be provided, with the data about the new server.
+   */
+  static processServerChange(
+    router: Router,
+    vpnClientService: VpnClientService,
+    vpnSavedDataService: VpnSavedDataService,
+    snackbarService: SnackbarService,
+    dialog: MatDialog,
+    dialogRef: MatDialogRef<any>,
+    localPk: string,
+    newServerFromHistory: LocalServerData,
+    newServerFromDiscovery: VpnServer,
+    newServerManually: ManualVpnServerData,
+    password: string,
+  ) {
+    // Check if the new server param was provided as it should.
+    let requestedPk: string;
+    if ((newServerFromHistory && (newServerFromDiscovery || newServerManually)) ||
+      (newServerFromDiscovery && (newServerFromHistory || newServerManually)) ||
+      (newServerManually && (newServerFromHistory || newServerFromDiscovery))
+    ) {
+      throw new Error('Invalid call');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skywire/pullRequest?prid=7163595)
```suggestion
      throw new Error("Invalid call");
```